controllers/controller.js.js

```diff
@@ -14,6 +14,21 @@ exports.getAllEndpoints = (req,res) => {
     res.status(200).send({ endPoints })
 }
 
+exports.getArticleById = (req,res,next) => {
+    const { article_id } = req.params
+    selectArticleById(article_id).then((article) => 
+    res.status(200).send( { article }))
+.catch((err) => {
+    if (err.status === 404) {
```

Comment:
You dont need to check the error here. You have set the status and message in your promise.reject. The error handler in your app.js should simply receive this and send it back to the user.